app/models/setting/discovered.rb

```diff
@@ -8,7 +8,9 @@ class Setting::Discovered < ::Setting
   BLANK_ATTRS << 'discovery_facts_hardware'
   BLANK_ATTRS << 'discovery_facts_network'
   BLANK_ATTRS << 'discovery_facts_ipmi'
-  BLANK_ATTRS << 'discovery_prefix'
+
+  HOSTNAME_ATTRS = ['discovery_hostname', 'discovery_prefix']
```

Comment:
Are these really attrs that contain a hostname? prefix suggests otherwise. Perhaps `STRING_PRESENCE_ATTRS` is closer to what it does?

Reply:
Done!